packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -3593,6 +3593,40 @@ const tests = {
         },
       ],
     },
+    {
+      code: normalizeIndent`
+        function MyComponent(props) {
+            function MyComponent(props) {
+                useEffect(() => {}, [props?.attribute.method()]);
+            }
+        }
+      `,
+      errors: [
+        {
+          message:
+            'React Hook useEffect has a complex expression in the dependency array. ' +
+            'Extract it to a separate variable so it can be statically checked.',
+          suggestions: undefined,
```

Comment:
The suggestion could just be a variable outside the hook

Reply:
Thanks for comment @reinaldosimoes. As this is just a bug fix to prevent rule from crashing the linter, I won't be making functionality changes. ESLint report is identical with old implementation without optional chaining:

```js
useEffect(() => {}, [someObject?.optionalField.method()]);
useEffect(() => {}, [someObject.method()]);

// Both produce:
// 'React Hook useEffect has a complex expression in the dependency array. Extract it to a separate variable so it can be statically checked.'
// -> No suggestion
```